sdk/keyvault/Azure.Security.KeyVault.Secrets.Extensions.Configuration/src/AzureKeyVaultConfigurationExtensions.cs

```diff
@@ -0,0 +1,131 @@
+            this IConfigurationBuilder configurationBuilder,
+            Uri vault,
+            IKeyVaultSecretManager manager)
+        {
+            return AddAzureKeyVault(configurationBuilder, vault, new DefaultAzureCredential(), manager);
+        }
+
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from the Azure KeyVault.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="vault">Azure KeyVault uri.</param>
+        /// <param name="credential">The credential to to use for authentication.</param>
+        /// <param name="manager">The <see cref="IKeyVaultSecretManager"/> instance used to control secret loading.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddAzureKeyVault(
+            this IConfigurationBuilder configurationBuilder,
+            Uri vault,
+            TokenCredential credential,
+            IKeyVaultSecretManager manager)
+        {
+            return AddAzureKeyVault(configurationBuilder, new AzureKeyVaultConfigurationOptions(vault, credential)
+            {
+                Manager = manager
+            });
+        }
+
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from the Azure KeyVault.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="client">The <see cref="SecretClient"/> to use for retrieving values.</param>
+        /// <param name="manager">The <see cref="IKeyVaultSecretManager"/> instance used to control secret loading.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddAzureKeyVault(
+            this IConfigurationBuilder configurationBuilder,
+            SecretClient client,
+            IKeyVaultSecretManager manager)
+        {
+            return configurationBuilder.Add(new AzureKeyVaultConfigurationSource(new AzureKeyVaultConfigurationOptions()
+            {
+                Client = client,
+                Manager = manager
+            }));
+        }
+
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from the Azure KeyVault.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="options">The <see cref="AzureKeyVaultConfigurationOptions"/> to use.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddAzureKeyVault(this IConfigurationBuilder configurationBuilder, AzureKeyVaultConfigurationOptions options)
+        {
+            if (configurationBuilder == null)
```

Comment:
Use the `Argument` shared class for consistency. Add a ProjectReference to the shared project.